packages/block-library/src/paragraph/edit.js

```diff
@@ -64,12 +64,26 @@ function ParagraphBlock( {
 	const blockProps = useBlockProps( {
 		ref: useOnEnter( { clientId, content } ),
 		className: classnames( {
-			'has-drop-cap': dropCap,
+			'has-drop-cap':
+				align === ( isRTL() ? 'left' : 'right' ) || align === 'center'
+					? false
+					: dropCap,
 			[ `has-text-align-${ align }` ]: align,
 		} ),
 		style: { direction },
 	} );
 
+	let helpText;
+	if ( align === ( isRTL() ? 'left' : 'right' ) || align === 'center' ) {
+		helpText = __( 'Aligned text can not have a drop cap.' );
+		// This attribute change also makes sure that the option is toggled off, not only disabled.
+		setAttributes( { dropCap: undefined } );
```

Comment:
>setAttributes( { dropCap: undefined } );

My gut says that this should be separate and not in the "help text" check, what do you think?

Reply:
That's correct. We shouldn't call `setAttribute` during the render.

Can you give me more details on when we want to set `dropCap` to `undefined`?

Reply:
If the drop cap is first enabled and then the text is aligned, the control is disabled.
The option should also be toggled off, not remain toggled on, because the drop cap is not used. 

This is what the control looks like if it is disabled but toggled on:
<img width="277" alt="Disabled toggle control with option enabled" src="https://user-images.githubusercontent.com/7422055/181166901-ca33267a-86f5-4e7c-a477-7d6f4faf5e6b.png">

I'll try to see if I can just put it on ToggleControl "checked".

Reply:
Thanks for the clarification.

I think this one change callback looks more suitable for setting the attribute.

https://github.com/WordPress/gutenberg/blob/9149f524eab181626531981f54ba1ca53447c231/packages/block-library/src/paragraph/edit.js#L92-L94

I also see that this check is repeated in a few places, so it might be good to extract in small function maybe `hasDropCap` or `hasDropCapDisabled`.

```js
// align === ( isRTL() ? 'left' : 'right' ) || align === 'center'
// Maybe something like this.
function hasDropCapDisabled( align ) {
    return align === ( isRTL() ? 'left' : 'right' ) || align === 'center';
}
```

Reply:
I found that when the dropCap attribute is only updated on change, there is no way to use the drop cap control to remove a previously existing drop cap if the language direction is changed.

If I use a rtl language, enable the drop cap, and then switch the language direction on the WP install to ltr, the `has-drop-cap` class is already on the paragraph. If I use the control to toggle the drop cap off and then on, the paragraph ends up having duplicate `has-drop-class `classes.
The class can be removed in the Advanced section, under "Additional CSS classes". 

Is this an edge case that can be ignored?

Reply:
> Is this an edge case that can be ignored?

I think so. Updating content/attributes based on user language preferences might be more confusing.